pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -402,6 +402,19 @@ func handleModify(ctx *downloaderContext, key string,
 	if config.RefCount != 0 && (status.HasError() || status.State != types.DOWNLOADED) {
 		log.Functionf("handleModify installing %s", config.Name)
 		handleCreate(ctx, config, status, key, receiveChan)
+		// Retrying the download due to image verification failure
+		// This happens when: RefCount and retryCount is non-zero,
+		// and DownloaderStatus state is "downloaded"
+	} else if config.RefCount != 0 && config.BlobDownloadRetryCount != 0 &&
+		status.BlobDownloadRetryCount != config.BlobDownloadRetryCount {
+		log.Functionf("handleModify retry download %s", config.Name)
+		status.CurrentSize = 0
+		status.Size = 0
+		status.Progress = 0
+		status.State = types.DOWNLOADING
+		status.BlobDownloadRetryCount++
```

Comment:
This is weird. If you set it to config.BlobDownloadRetryCount I could understand it.

Reply:
Yep, I have the same understanding. If we take a counter approach, we only need to store the config value here.

Reply:
Changed the BlobDownloadRetryCount to LastRetry. I like this approach better.